src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceService.kt

```diff
@@ -49,9 +42,9 @@ class LicenceService(
     val isCas2Accepted = cas2Referral?.bassOffer?.bassAccepted == OfferAccepted.YES
 
     val address = when {
-      isCurfewApprovedPremisesRequired && !isCas2ApprovedPremisesRequired -> curfew.approvedPremisesAddress
-      isCas2ApprovedPremisesRequired -> cas2Referral.approvedPremisesAddress
-      isCas2Requested && isCas2Accepted -> cas2Referral.bassOffer
+      isCurfewApprovedPremisesRequired && !isCas2ApprovedPremisesRequired -> curfew?.approvedPremisesAddress
```

Comment:
mmh.. these `?` shouldn't be needed as smart casting should detect that it's not possible for them to null? 
<img width="1027" alt="Screenshot 2025-01-09 at 18 14 03" src="https://github.com/user-attachments/assets/2a5545a9-c8c8-426f-b072-e1a7d0459770" />

Reply:
I'm still getting swiggles despite trying everything but removing them seems to be fine too.